aptos-move/framework/aptos-framework/sources/simple_delegations.move

```diff
@@ -0,0 +1,844 @@
+    public fun get_stake_pool_address(delegator: address, operator: address): address acquires Delegations {
+        let delegations = &borrow_global<Delegations>(delegator).delegations;
+        simple_map::borrow(delegations, &operator).pool_address
+    }
+
+    public fun get_last_recorded_total_stake(delegator: address, operator: address): u64 acquires Delegations {
+        let delegations = &borrow_global<Delegations>(delegator).delegations;
+        simple_map::borrow(delegations, &operator).total_stake
+    }
+
+    public fun get_commission_percentage(delegator: address, operator: address): u64 acquires Delegations {
+        let delegations = &borrow_global<Delegations>(delegator).delegations;
+        simple_map::borrow(delegations, &operator).commission_percentage
+    }
+
+    public fun is_delegation_active(delegator: address, operator: address): bool acquires Delegations {
+        let delegations = &borrow_global<Delegations>(delegator).delegations;
+        simple_map::borrow(delegations, &operator).is_active
+    }
+
+    /// Delegator can call this function to define a simple delegation contract with a specified operator.
+    ///
+    /// Can only delegate to a specific operator once. Afterward, delegator cannot update amount or commission.
+    public entry fun delegate(
+        delegator: &signer,
+        operator: address,
+        voter: address,
+        amount: u64,
+        commission_percentage: u64,
+    ) acquires Delegations {
+        assert!(
+            commission_percentage >= 0 && commission_percentage <= 100,
+            error::invalid_argument(EINVALID_COMMISSION_PERCENTAGE),
+        );
+        // The delegated stake should be at least the min_stake_required, so the stake pool will be eligible to join the
+        // validator set.
+        let (min_stake_required, _) = staking_config::get_required_stake(&staking_config::get());
+        assert!(amount > min_stake_required, error::invalid_argument(EINSUFFICIENT_DELEGATION_AMOUNT));
+
+        // Initialize Delegations resource if this is the first the delegator has delegated to anyone.
+        let delegator_address = signer::address_of(delegator);
+        if (!exists<Delegations>(delegator_address)) {
+            move_to(delegator, new_delegations_resource(delegator));
+        };
+
+        // Only allow delegating to the same operator once.
+        let delegations = borrow_global_mut<Delegations>(delegator_address);
+        let delegations_map = &mut delegations.delegations;
+        assert!(
+            !simple_map::contains_key(delegations_map, &operator),
+            error::invalid_argument(ECANNOT_CHANGE_EXISTING_DELEGATION)
+        );
+
+        // Initialize the stake pool in a new resource account. This allows the same delegator to delegate to multiple
+        // different operators.
+        let seed = create_seed(delegator_address, operator);
+        let (stake_pool_signer, _) = account::create_resource_account(delegator, seed);
```

Comment:
should we also store the signer capability just in case?

Reply:
I'm not sure if that would be useful at all as stake is written to allow OwnerCapability to 100% control the stake pool. There should be no owner-only functions in the stake module that can be called with a signer and not have a with_owner_cap version. If there's any, it's a bug we need to fix.

Reply:
It's always better to be safe than sorry, imagine if there's a bug and we lose the signer capability and there's no easy way to fix it

Reply:
Okay I added signer_cap to the Delegations resource. We won't need it immediately but can later upgrade (subject to onchain gov) if needed to use it.